cli/klaus/src/main.rs

```diff
@@ -99,6 +100,21 @@ fn main() -> Result<(), Error> {
     #[cfg(target_os = "linux")]
     nix::sys::prctl::set_pdeathsig(Some(nix::sys::signal::Signal::SIGUSR1))
         .context("set_pdeathsig failed")?;
+    let ppid = getppid();
+    // Check if our original parent has already exited
+    // Even if our ppid has changed since the prctl, we still know our parent
+    // has exited, so it's okay to cleanup.
+    if ppid.as_raw() != args.pid {
+        debug!("parent has already exited");
+        let reg = read_environment_registry(&args.registry_path)?.unwrap_or_default();
+        let entry = reg
+            .entry_for_hash(&args.dot_flox_hash)
+            .ok_or(EnvRegistryError::UnknownKey(args.dot_flox_hash))?;
+        if entry.activations.is_empty() {
+            cleanup(args.socket_path);
+        }
+        return Ok(());
+    }
 
     #[cfg(target_os = "macos")]
```

Comment:
@zmitchell do you know if kqueue will still send us an event if the PID has already exited? I'm thinking it might be based on this:
```
     The filter identified in a kevent is executed upon the initial registration of that event in order to detect whether a preexisting condition is present, and is also executed whenever
     an event is passed to the filter for evaluation.  If the filter determines that the condition should be reported, then the kevent is placed on the kqueue for the user to retrieve.
```